CompressImagesFunction/CompressImages.cs

```diff
@@ -131,16 +131,52 @@ public static bool Run(CompressimagesParameters parameters, ICollector<CompressI
 
             // reset any mean files
             repo.Reset(ResetMode.Mixed, repo.Head.Tip);
-
+            Console.WriteLine("flerb");
             // optimize images
             string[] imagePaths;
+            List<string> addedOrModifiedImagePaths = new List<string>();
+            List<string> deletedImagePaths = new List<string>();
             if (parameters.IsRebase)
             {
-                // TODO: find conflicting images and new images :)
-                imagePaths = new string[] { parameters.LocalPath + "/1.png" };
+                Console.WriteLine("This is rebase");
+                //TODO: find conflicting images https://libgit2.org/libgit2/#HEAD/group/diff/git_diff_foreach
+            
+                var refspec = string.Format("{0}:{0}", KnownGitHubs.BranchName);
+                Commands.Fetch(repo, "origin", new List<string> { refspec }, null, "fetch");
+
+                Console.WriteLine("Diff: repo.Branches[KnownGitHubs.BranchName].Tip.Sha: " + repo.Branches[KnownGitHubs.BranchName].Tip.Sha + "repo.Head.Tip.Sha: " +  repo.Head.Tip.Sha);
+                var diff = repo.Diff.Compare<TreeChanges>(repo.Branches[KnownGitHubs.BranchName].Tip.Tree, repo.Head.Tip.Tree);
+
+                if (diff != null) 
```

Comment:
It's very helpful to check if the diff is null, because then we know right away if the rebase does not make sense. But they use a "golden path" style so we should do the null-check first, and returning on failure. More or less like this:

```
if(diff == null){
  //log.info("rebase failed or something")
  return false;
}
//continue here with correct behavior
...
```